ws-server/src/server.rs

```diff
@@ -124,40 +124,49 @@ async fn background_task(
 		}
 	});
 
+	// Buffer for incoming data.
 	let mut data = Vec::with_capacity(100);
 
-	// Look up the "method" (i.e. function pointer) from the registered methods and run it passing in
-	// the params from the request. The result of the computation is sent back over the `tx` channel and
-	// the result(s) are collected into a `String` and sent back over the wire.
+	// Look up the "method" (i.e. function pointer) from the registered methods and run it passing in the params from
+	// the request. The result of the computation is sent back over the `tx` channel and the result(s) are collected
+	// into a `String` and sent back over the wire.
 	//
-	// Note: This handler expects method existence to be checked prior to the call and will panic if
+	// Note: This handler expects method existence to be checked prior to starting the server and will panic if the
 	// method does not exist.
 	let sync_methods = methods.clone();
 	let execute_sync = move |tx: &MethodSink, req: JsonRpcRequest| {
-		let method = sync_methods.method(&*req.method).unwrap();
+		let method = sync_methods
+			.sync_method(&*req.method)
+			.expect(&format!("sync method '{}' is not registered on the server – this is a bug", req.method));
```

Comment:
unwrap_or_else(|| panic!("sync method '{}' is not registered on the server – this is a bug", req.method));